File: LeetCode_Marathons/LCM1/Easy/Is_Subsequence/a.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        indexS, indexT = 0, 0

        while indexS != len(s) and indexT != len(t):
            if s[indexS] == t[indexT]: indexS += 1
            indexT += 1

        return indexS == len(s)

File: LeetCode_Marathons/LCM1/Easy/Is_Subsequence/a.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int indexS = 0, indexT = 0;

        while (indexS != s.length() && indexT != t.length()) {
            if (s.charAt(indexS) == t.charAt(indexT)) indexS++;
            indexT++;
        }
        
        return indexS == s.length();
    }
}